crates/pathfinder/src/storage/state.rs

```diff
@@ -210,29 +196,18 @@ impl RefsTable {
     /// Returns the current L1-L2 head. This indicates the latest block for which L1 and L2 agree.
     pub fn get_l1_l2_head(tx: &Transaction<'_>) -> anyhow::Result<Option<StarknetBlockNumber>> {
         // This table always contains exactly one row.
-        let block_number =
-            tx.query_row("SELECT l1_l2_head FROM refs WHERE idx = 1", [], |row| {
-                let block_number = row
-                    .get_ref_unwrap(0)
-                    .as_i64_or_null()
-                    .unwrap()
-                    .map(|x| StarknetBlockNumber(x as u64));
-
-                Ok(block_number)
-            })?;
-
-        Ok(block_number)
+        tx.query_row("SELECT l1_l2_head FROM refs WHERE idx = 1", [], |row| {
+            row.get::<_, Option<_>>(0)
```

Comment:
This was a peculiar one, obvious in hindsight. To get an **optional column value** then you must get an option out of the **row**.